src/store/comment.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * @copyright Copyright (c) 2018 Julius Härtl <jus@bitgrid.net>
+ *
+ * @author Julius Härtl <jus@bitgrid.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { CommentApi } from '../services/CommentApi'
+import xmlToTagList from '../helpers/xml'
+import Vue from 'vue'
+
+const apiClient = new CommentApi()
+
+export default {
+	state: {
+		comments: {}
+	},
+	mutations: {
+		addComments(state, commentObj) {
+			if (state.comments[commentObj.cardId] === undefined) {
+				Vue.set(state.comments, commentObj.cardId, commentObj.comments)
+			} else {
+				// FIXME append comments once incremental fetching is implemented
```

Comment:
This needs some more adjustments, The Vue.set was just a way to avoid duplicate entries when testing. We need to add elements to the existing list, but only if they are not already part of it.

Reply:
How can I use set with these complex command?
state.comments[comment.cardId][existingIndex].message = comment.comment

This is not working
```
Vue.set(state.comments, comment.cardId.existingIndex.message, comment.comment)
Vue.set(state.comments, comment.cardId[existingIndex].message, comment.comment)
Vue.set(state.comments[comment.cardId][existingIndex], message, comment.comment)
```